git-tar/function/ops.go

```diff
@@ -209,9 +236,100 @@ func deploy(tars []tarEntry, pushEvent sdk.PushEvent, stack *stack.Services) err
 		res, reqErr := c.Do(httpReq)
 		if reqErr != nil {
 			fmt.Fprintf(os.Stderr, fmt.Errorf("unable to deploy function via buildshiprun: %s", reqErr.Error()).Error())
+			hasFailure = true
+			failedFunctions[tarEntry.functionName] = reqErr.Error()
 		}
 
 		fmt.Println("Service deployed ", tarEntry.functionName, res.Status, owner)
 	}
+
+	if !hasFailure {
+		reportStatus("success", fmt.Sprintf("service successfully deployed as: %s", serviceValue), "DEPLOY", event)
+	} else {
+		reportStatus("failure", fmt.Sprintf("failed to deploy service %s: %v", serviceValue, failedFunctions), "DEPLOY", event)
```

Comment:
We can't report status on a single service/function - we have to do it for the whole stack as a yes/no. Why? Because what if one passes then the following fails and the one after that passes? The final status will say passed and hide the error..

Reply:
As I understand if the stack has one or more failure the status should be a failure. Currently `hasFailure` is `false` by default, it's being set to `true` only when a function fails, and then there is no way it can be set to `false`.  That means if there is one function failure the whole stack would be `failure`, which I think is intended. 

I have added a `failedFunctions map[string]string` which would map `functionname` -> `error` for failed functions